main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -1705,11 +1720,22 @@ object Namer extends Phase[WeededAst.Program, NamedAst.Root] {
   /**
     * Merges the given `uses` into the given use environment `uenv0`.
     */
-  private def mergeUseEnvs(uses: List[WeededAst.Use], uenv0: UseEnv): Validation[UseEnv, NameError] =
+  private def mergeUseEnvs(uses: List[WeededAst.Use], uenv0: UseEnv): Validation[UseEnv, NameError] = {
+
+    def locationOfUseDefOrSig(name: String, uenv: UseEnv): Option[SourceLocation] = {
+      uenv.defs.get(name) match {
+        case None => uenv.sigs.get(name) match {
+          case Some((_clazz, ident)) => Some(ident.loc)
+          case None => None
+        }
+        case Some(defn) => Some(defn.loc)
+      }
+    }
+
     Validation.fold(uses, uenv0) {
       case (uenv1, WeededAst.Use.UseClass(qname, alias, _)) =>
         val name = alias.name
-        uenv1.classes.get(name) match {
+        uenv1.classes.get(name) match { // MATT need to check against types as well once `use class` is changed.
```

Comment:
What is this?

Reply:
I was thinking that we wouldn't want type names and class names to conflict, but given that they're used differently syntactically, the ambiguity there shouldn't be an issue.

On the other hand, since a class `use` and type `use` can look the same, we'd need to make sure at the point of declaration that there's no conflict:

```flix
namespace N {
    class C[a]
    type alias C = Int
}
...
use N.C; // type or class?
```

That said, there may be an issue with signature/definition ambiguity at top level classes vs namespaces:

```flix
namespace N {
    def f(): Int = 123
}
class N {
    def f(): Int
}
...
use N.f; // sig or def?
```

I'll need to look into this.

Reply:
> I'll need to look into this.

(Closely related to other comment)